test/e2e/Tests/Helpers/DurableHelpers.cs

```diff
@@ -52,6 +52,21 @@ internal static async Task<OrchestrationStatusDetails> GetRunningOrchestrationDe
         return new OrchestrationStatusDetails(statusQueryResponseString);
     }
 
+    internal static async Task WaitForOrchestrationStateAsync(string statusQueryGetUri, string desiredState, int maxTimeoutSeconds)
+    {
+        DateTime timeoutTime = DateTime.Now + TimeSpan.FromSeconds(maxTimeoutSeconds);
+        while (DateTime.Now < timeoutTime)
+        {
+            var currentStatus = await GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+            if (currentStatus.RuntimeStatus == desiredState)
+            {
+                return;
+            }
```

Comment:
As discussed, let's also handle the case where the orchestration has completed/terminated/failed, in which case additional polling would be futile.  Something like this:
```suggestion
            }
            else if (currentStatus.RuntimeStatus == "Completed" || currentStatus.RuntimeStatus == "Terminated" || currentStatus.RuntimeStatus == "Failed")
            {
                throw new InvalidOperationException($"The orchestration ended unexpectedly in a {currentStatus.RuntimeStatus} state!");
            }
```

Reply:
I did add this, on the next lines
```C#
            if (finalStates.Contains(currentStatus.RuntimeStatus))
            {
                throw new TaskCanceledException($"Orchestration reached {currentStatus.RuntimeStatus} state when test was expecting {desiredState}");
            }
```